src/sage/plot/graphics.py

```diff
@@ -432,6 +432,19 @@ def set_legend_options(self, **kwds):
 
             sage: p.set_legend_options(loc=(0.5,0.5)); p  # aligns the bottom of the box to the center                  # needs sage.symbolic
             Graphics object consisting of 1 graphics primitive
+    
+        The dictionary ``_extra_kwds`` stores the keywords for graphics. This can be used to adjust vertical or
+        horizonal displacements from the x-axis or y-axis respectively::
+        
+            sage: p = line([(0, 0), (1, 1)], legend_label='test', axes_labels=('X LABEL', 'Y LABEL'))
+            sage: labels = p._extra_kwds.get('axes_labels')
+            sage: if labels and labels[0]:
+            ....:   displace = -.14 - .005 * p.fontsize()
+            ....: else:
+            ....:   displace = -.14
+            ....: p.set_legend_options(bbox_to_anchor=(0., displace, 1., .102), loc=3, mode="expand", borderaxespad=0.)
```

Comment:
You shouldn't need to go through this kind of code for this example. The value of `displace` should be explicitly known in this example.

Also, use `0.0` instead of `0.`.

This also seems to be using options that are not documented (`bbox_to_anchor`, `mode`). These should be added.